cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2630,46 +2628,101 @@ macro(build_awssdk)
 
   externalproject_add(awssdk_ep
                       ${EP_LOG_OPTIONS}
                       URL ${AWSSDK_SOURCE_URL}
                       CMAKE_ARGS ${AWSSDK_CMAKE_ARGS}
-                      BUILD_BYPRODUCTS ${AWSSDK_SHARED_LIBS})
+                      BUILD_BYPRODUCTS ${AWSSDK_STATIC_LIBS})
 
   file(MAKE_DIRECTORY ${AWSSDK_INCLUDE_DIR})
 
+  add_library(AWS::aws-checksums STATIC IMPORTED)
+  set_target_properties(AWS::aws-checksums
+                        PROPERTIES IMPORTED_LOCATION "${AWSSDK_CHECKSUMS_STATIC_LIB}"
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   "${AWSSDK_INCLUDE_DIR}")
+
+  add_library(AWS::aws-c-common STATIC IMPORTED)
+  set_target_properties(AWS::aws-c-common
+                        PROPERTIES IMPORTED_LOCATION "${AWSSDK_COMMON_STATIC_LIB}"
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   "${AWSSDK_INCLUDE_DIR}")
+
+  add_library(AWS::aws-c-event-stream STATIC IMPORTED)
+  set_target_properties(AWS::aws-c-event-stream
+                        PROPERTIES IMPORTED_LOCATION "${AWSSDK_ES_STATIC_LIB}"
+                                  INTERFACE_INCLUDE_DIRECTORIES
+                                  "${AWSSDK_INCLUDE_DIR}")
+
+  add_library(AWS::aws-cpp-sdk-core STATIC IMPORTED)
+  set_target_properties(AWS::aws-cpp-sdk-core
+                        PROPERTIES IMPORTED_LOCATION "${AWSSDK_CORE_STATIC_LIB}"
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   "${AWSSDK_INCLUDE_DIR}")
+
+  add_library(AWS::aws-cpp-sdk-s3 STATIC IMPORTED)
+  set_target_properties(AWS::aws-cpp-sdk-s3
+                        PROPERTIES IMPORTED_LOCATION "${AWSSDK_S3_STATIC_LIB}"
+                                  INTERFACE_INCLUDE_DIRECTORIES
+                                  "${AWSSDK_INCLUDE_DIR}")
+
+  add_library(AWS::aws-cpp-sdk-cognito-identity STATIC IMPORTED)
+  set_target_properties(AWS::aws-cpp-sdk-cognito-identity
+                        PROPERTIES IMPORTED_LOCATION "${AWSSDK_COG_STATIC_LIB}"
+                                  INTERFACE_INCLUDE_DIRECTORIES
+                                  "${AWSSDK_INCLUDE_DIR}")
+
+  add_library(AWS::aws-cpp-sdk-identity-management STATIC IMPORTED)
+  set_target_properties(AWS::aws-cpp-sdk-identity-management
+                        PROPERTIES IMPORTED_LOCATION "${AWSSDK_IAM_STATIC_LIB}"
+                                  INTERFACE_INCLUDE_DIRECTORIES
+                                  "${AWSSDK_INCLUDE_DIR}")
+
+  add_library(AWS::aws-cpp-sdk-sts STATIC IMPORTED)
+  set_target_properties(AWS::aws-cpp-sdk-sts
+                        PROPERTIES IMPORTED_LOCATION "${AWSSDK_STS_STATIC_LIB}"
+                                  INTERFACE_INCLUDE_DIRECTORIES
+                                  "${AWSSDK_INCLUDE_DIR}")
+
+  # on linux and macos curl seems to be required
+  find_package(CURL REQUIRED)
```

Comment:
I'm not sure how feasible is this, perhaps we can turn off the curl dependency of AWS. I assume linking against the system curl should be fine rather than compiling curl as well.